csync/src/std/c_string.c

```diff
@@ -274,13 +274,72 @@ char* c_utf8_from_locale(const mbchar_t *wstr)
   return dst;
 }
 
+/*
+ * This function takes a path and converts it to a UNC representation of the
+ * string. That means that it prepends a \\?\ and convertes all slashes to
+ * backslashes.
+ *
+ * Note the following:
+ *  - The string must be absolute.
+ *  - it needs to contain a drive character to be a valid UNC
+ *  - A conversion is only done if the path len is larger than 245. Otherwise
+ *    the windows API functions work with the normal "unixoid" representation too.
+ *
+ * Since the function reallocs memory that it can not free itself, the number of
+ * newly allocated bytes are returned in parameter mem_reserved. The calling
+ * function will call free on the result pointer if mem_reserved is > 0.
+ *
+ */
+ const char *makeLongWinPath(const char *str, int *mem_reserved)
```

Comment:
I think this function should not be in c_string,  but more like somewhere in vio.  It is not a string related function  (not something you would see in QString or QByteArray)  but a function related to path, so more like something you would find in QDir or QFile


Anyway, the real problem of the function is the allocation behaviour.  it sometimes allocate and sometimes not.   and that is i think a problem.